### PR TITLE
snforge_std: remove `RevertedTransaction`

### LINKED ISSUE #1887: `RevertedTransaction` remake
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

We should:
- remove `.first()` method (now pretty useless since there are bytearrays)
- mapping error trait remake:
  - (`.map_string_error()`) like in syscall result (and rename the trait to be more generic)
  - rename above function to be `map_error_to_string()`

### PR DESCRIPTION
**Pull request by drknzz**
Closes #1887

## Introduced changes

- remove `RevertedTransaction`
- L1Handler `execute()` returns `SyscallResult<()>`
- `declare()` returns `Result<ContractClass, Array<felt252>>`

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: snforge_std/src/cheatcodes/contract_class.cairo [hunk 1]
```diff
@@ -2,21 +2,6 @@ use starknet::{ContractAddress, ClassHash, testing::cheatcode, SyscallResult};
...
 use core::traits::Into;
 
-#[derive(Drop, Clone)]
-struct RevertedTransaction {
-    panic_data: Array::<felt252>,
-}
-
-trait RevertedTransactionTrait {
-    fn first(self: @RevertedTransaction) -> felt252;
-}
```
Comment (Arcticae) [inline snforge_std/src/cheatcodes/contract_class.cairo:12]: Wasn't this documented? I don't see any deletions of that from the docs

Comment (drknzz) [reply snforge_std/src/cheatcodes/contract_class.cairo:12]: Not separately as a trait. Only mentioned in the return types.

**Review by piotmag769 — APPROVED** [review]
wrong base branch or the base is not rebased

**Review by piotmag769 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,7 +27,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 #### Changed
 
 - `read_txt` and `read_json` now supports `ByteArray`
-- `snforge_std::declare` returns `Result<ContractClass, RevertedTransaction>` instead of `ContractClass`
```
[the remark this answers is not in the document]
Comment (piotmag769) [reply CHANGELOG.md:24]: We shouldn't modify our historic changelog :/

[the remark this answers is not in the document]
Comment (drknzz) [reply CHANGELOG.md:24]: Agree.
This hasn't been released though, see [v0.21.0 CHANGELOG](https://github.com/foundry-rs/starknet-foundry/blob/v0.21.0/CHANGELOG.md)

**Pull request merged**